csound-ctcsound.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/interfaces/CMakeLists.txt b/interfaces/CMakeLists.txt
+index aff6e8fec..5ab32991e 100644
+--- a/interfaces/CMakeLists.txt
++++ b/interfaces/CMakeLists.txt
+@@ -112,7 +112,7 @@ LUA_LIBRARY LUA_H_PATH)
+ if(NOT APPLE)
+       execute_process (
+            COMMAND python3 -c
+-           "import site, sys; sys.stdout.write(site.getsitepackages()[0])"
++           "import site, sys; sys.stdout.write(site.getsitepackages()[0].replace('usr', 'app'))"
```

Comment:
For the sake of correctness, can you investigate how `site.getsitepackages()` works? It might lead us to a proper solution, instead of this patch.

Reply:
I saw that `getsitepackages()` returns a list of all global site-packages, testing inside flatpak filesystem:
\> `['/usr/lib/python3.10/site-packages']`
There is also `getusersitepackages()` which returns the path of the user-specific site-packages directory
\> `'/var/data/python/lib/python3.10/site-packages'`

There is also `python -m site` which has the path we want under /app
```
[📦 org.sugarlabs.MusicKeyboard ~]$ python -m site
sys.path = [
    '/home/sparsh',
    '/usr/lib/python310.zip',
    '/usr/lib/python3.10',
    '/usr/lib/python3.10/lib-dynload',
    '/app/lib/python3.10/site-packages',
    '/usr/lib/python3.10/site-packages',
]
USER_BASE: '/home/sparsh/.local' (exists)
USER_SITE: '/home/sparsh/.local/lib/python3.10/site-packages' (doesn't exist)
ENABLE_USER_SITE: True
```

We will have to make a patch anyways i guess

Reply:
Isn't there an environment variable that we can override so that `getsitepackages` only returns the path we want?

Reply:
hmm I figured out that we can do `export PYTHONUSERBASE="/app"` and then we get `site.getusersitepackages()` under /app as we want, but `site.getsitepackages()` still points towards the global path. I don't see any relevant env var we can set in [docs](https://docs.python.org/3/library/site.html) to change the global site path, maybe we can somehow override `site.getsitepackages()`? but idk if that would be a good idea.

Reply:
Please investigate, and determine if there's an environment variable under `getsitepackages`.

Reply:
@sparshg check the actual code of `getsitepackages`.

Reply:
Okay so going through the code I see the path actually comes from `sys.prefix` which is determined by `sys.path` which is determined either automatically or through `PYTHONHOME` and `PYTHONPATH` if set. I finally got it working with:

`PYTHONHOME=/app PYTHONPATH=/usr/lib/python3.10 python`
```json
"env": {
    "PYTHONHOME": "/app",
    "PYTHONPATH": "/usr/lib/python3.10"
}
```
I will try adding this in build-options and it should work but we are hardcoding version 3.10 into the env variable, is this right?